src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuIndirectCostRecoveryAccountDistributionMetadata.java

```diff
@@ -0,0 +1,76 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.cornell.kfs.gl.batch.service.impl;
+
+import java.text.MessageFormat;
+
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.businessobject.IndirectCostRecoveryAccount;
+import org.kuali.kfs.gl.batch.service.impl.IndirectCostRecoveryAccountDistributionMetadata;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+
+public class CuIndirectCostRecoveryAccountDistributionMetadata extends IndirectCostRecoveryAccountDistributionMetadata {
+
+    /**
+     * @param icrAccount
+     * @throws Exception
+     */
+    public CuIndirectCostRecoveryAccountDistributionMetadata(IndirectCostRecoveryAccount icrAccount) throws UnsupportedOperationException {
+      super(icrAccount);
+
+      // Do what super does if the ICRA is null or not closed
+      if(ObjectUtils.isNotNull(icrAccount) && 
+         ObjectUtils.isNotNull(icrAccount.getIndirectCostRecoveryAccount()) && 
+         icrAccount.getIndirectCostRecoveryAccount().isClosed()) {
+        Account replacementICRA = icrAccount.getIndirectCostRecoveryAccount();
+        int i = 0;
+
+        // Search 10 levels or less to find an open Account, as we do in other processes
+        while((i < 10) && ((replacementICRA != null) && replacementICRA.isClosed())) {
+          replacementICRA = replacementICRA.getContinuationAccount();
+          i++;
+        }
+  
+        // If we've found a replacement that isn't what we've already got...
+        if((replacementICRA != null) && (replacementICRA != icrAccount.getIndirectCostRecoveryAccount())) {
+          if(replacementICRA.isClosed()) {
+            // If the ICR Account is still closed after searching this far down, raise an exception.
+            String errorText = SpringContext.getBean(ConfigurationService.class)
+                                            .getPropertyValueAsString(CUKFSKeyConstants.ERROR_ICRACCOUNT_CONTINUATION_ACCOUNT_CLOSED);
+            Object[] args = {
+                icrAccount.getIndirectCostRecoveryFinCoaCode(),
+                icrAccount.getIndirectCostRecoveryAccountNumber()
+              };
+            errorText = MessageFormat.format(errorText, args);
+  
+            throw new UnsupportedOperationException(errorText);
```

Comment:
As I have mentioned to you on Friday, I believe on hipchat, it does not seem right to throw this exception if an open ICR account is not found. If this was Dave's requirement then this is fine, otherwise I wonder if an error message should be written to the error report instead? I know you mentioned it is highly unlikely for this situation to occur but should this occur the poster job would fail with an exception and I am not sure this is the desired behavior.